src/libraries/Microsoft.PowerFx.Core/Texl/Builtins/AddColumns.cs

```diff
@@ -86,33 +88,59 @@ public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType
                 // arbitrary expressions that evaluate to string values, because these values contribute to
                 // type analysis, so they need to be known upfront (before AddColumns executes).
-                StrLitNode nameNode;
-                if (nameArgType.Kind != DKind.String ||
-                    (nameNode = nameArg.AsStrLit()) == null)
+                if (supportIndentifiers)
                 {
-                    fArgsValid = false;
-                    errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrExpectedStringLiteralArg_Name, nameArg.ToString());
-                    continue;
+                    FirstNameNode identifierNode = nameArg.AsFirstName();
+
+                    if (identifierNode == null)
+                    {
+                        fArgsValid = false;
+
+                        // Argument '{0}' is invalid, expected a text literal.
+                        errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrExpectedStringLiteralArg_Name, nameArg.ToString());
+                        continue;
+                    }
+
+                    value = identifierNode.Ident.Name;
+                }
+                else
+                {
+                    StrLitNode strLitNode = nameArg.AsStrLit();
+
+                    if (nameArgType.Kind != DKind.String && strLitNode == null)
+                    {
+                        fArgsValid = false;
+
+                        // Argument '{0}' is invalid, expected a text literal.
+                        errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrExpectedStringLiteralArg_Name, nameArg.ToString());
+                        continue;
+                    }
+
+                    value = strLitNode.Value;
                 }
 
                 // Verify that the name is valid.
-                if (!DName.IsValidDName(nameNode.Value))
+                if (!DName.IsValidDName(value))
                 {
                     fArgsValid = false;
-                    errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrArgNotAValidIdentifier_Name, nameNode.Value);
+
+                    // Argument '{0}' is not a valid identifier.
+                    errors.EnsureError(DocumentErrorSeverity.Severe, nameArg, TexlStrings.ErrArgNotAValidIdentifier_Name, value);
                     continue;
                 }
 
-                var columnName = new DName(nameNode.Value);
+                var columnName = new DName(value);
                 if (DType.TryGetDisplayNameForColumn(typeScope, columnName, out var colName))
                 {
                     columnName = new DName(colName);
                 }
 
                 // Verify that the name doesn't already exist as either a logical or display name
-                if (typeScope.TryGetType(columnName, out var columnType) || DType.TryGetLogicalNameForColumn(typeScope, columnName, out var unused))
```

Comment:
>|| DType.TryGetLogicalNameForColumn(typeScope, columnName, out var unused)) [](http://example.com/codeflow?start=73&length=75)

Why are we removing this collision check?

Reply:
Good catch.
I can't explain...